### ISSUE TITLE
Unexpected results for font-display ('Ensure text remains visible during webfont load') audit

### ISSUE THREAD

**Issue by erlapso** [GoogleChrome/lighthouse#6628]
I adjusted all the ‘font-display’ so that is optional or fallback as explained in the documentation, but I still get ‘Ensure text remains visible during webfont load‘ performance warning

**Comment by wardpeet** [comment]
@erlapso thanks for reporting! Feel free to correct me if I'm wrong. 👍 

There is something weird going on between pagespeedinsights and lighthouse.
pagespeedinsights:
![image](https://user-images.githubusercontent.com/1120926/48872360-09a87d00-ede9-11e8-8f0d-40358eec4d82.png)

lighthouse latest master seems to pass
![image](https://user-images.githubusercontent.com/1120926/48872459-6efc6e00-ede9-11e8-8fa7-d2eb8cea4dc7.png)

lighthouse devtools isn't: (v3.0.3)
![image](https://user-images.githubusercontent.com/1120926/48872508-a539ed80-ede9-11e8-8413-27912eee4e18.png)

Actually devtools v3.0.3 is correct 👀 . The site is using critical-css which is great! 🎉 I see lots of font-face declarations in here without font-display. When the stylesheets kick in the font-faces are loaded again which do have a font-display value. Does this make sense?

**Comment by blue2blond** [comment]
Any news on this? I am experiencing the same. Tried different font-display options als wel als putting it before or after src-property.

version: Lighthouse 4.0.0-alpha.1
url: blue2blond [dot] nl/sitemap.html
![lighthouse-font-display](https://user-images.githubusercontent.com/28271831/49877416-aa58ee00-fe25-11e8-85fb-d820010b7eb5.png)

**Comment by patrickhulce** [comment]
Ah thanks @blue2blond your issue is a little bit different, and I can confirm the problem. I'll file a separate issue for this 👍

**Comment by paulirish** [comment]
Also reported with bombacarta DOT com

**Comment by bhargavpjoshi** [comment]
Any further development in this thread??? Facing the same issue after deploying all available suggestions from my many of websites.???

for ref. https://www.truecadd.com/

**Comment by patrickhulce** [comment]
Current state of this issue:

- https://botsociety.io - passes locally on master, fails on web.dev
- https://www.truecadd.com/ - passes locally on master, passes on web.dev
- https://bombacarta.com/ - passes locally on master, fails on web.dev
- http://blue2blond.nl/sitemap.html - passes locally on master after #6782

I'm going to reassign this to @exterkamp for now as it's a LR-specific issue (seems like it'd be difficult for ward to figure this one out without LR debug access, but feel free to pass on to the appropriate person @exterkamp  😃). 

@TheDutchCoder if you have a URL for us to add to this list while debugging that'd be great!

**Comment by awran5** [comment]
Same issue here: https://gkstyle.net/

Passes locally, fails on PageSpeed Insights.

**Comment by Diklla** [comment]
Not sure this is the reason for all the above failures, but I had a similar issue, so I checked and turns out that the code that Lighthouse uses to find a declared font-display property is `declaration.match(/font-display:(.*?);/);`. Please note the semicolon at the end of the regular expression. This means that if font-display is the last line in the CSS rule and does not end with a semicolon, the diagnostic audit fails. Given that this is a rather common practice, it might explain some of the above fails and should be addressed regardless.

**Comment by patrickhulce** [comment]
Thanks very much @Diklla! Interested in picking up a PR for that? :)

**Comment by evromalarkey** [comment]
@Diklla good find, adding semicolon at the end of the line fixed the issue for me.

**Comment by awran5** [comment]
@Diklla Thank you very much! that fixed it for me too.

**Comment by paulirish** [comment]
fixed by #7191

**Comment by simeranya** [comment]
same problem is here: https://hastadoktor.com 

can you explain the answer for non-coder ?

**Comment by maersu** [comment]
Did have the same issue with `font-display: auto;`

Looking at the [lighthouse code](https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-core/audits/font-display.js#L10) shows that only `(block|fallback|optional|swap)` are valid options. 

Any reason why `auto` is missing? 

According to the [Google docs](https://developers.google.com/web/updates/2016/02/font-display) and [Mozilla docs](https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face/font-display) auto is `valid` option

**Comment by patrickhulce** [comment]
@maersu the audit is all about picking a behavior *other* than the default behavior. It's not about setting *any* valid value.

Ideally use `optional` or `swap` but there are certain use cases (font icons for example) where the others make sense.

**Comment by maersu** [comment]
@patrickhulce thanks for the reply and clarification! 

I was not aware, that `auto` is the default. 
IMHO it's a bit confusing since the ("Learn more" link)[https://developers.google.com/web/updates/2016/02/font-display] in the test does not mention it and `auto` is the example config there:

```
@font-face {
  font-family: 'Arvo';
  font-display: auto;
  src: local('Arvo'), url(https://fonts.gstatic.com/s/arvo/v9/rC7kKhY-eUDY-ucISTIf5PesZW2xOQ-xsNqO47m55DA.woff2) format('woff2');
}
```

**Comment by patrickhulce** [comment]
Hmm, you're right we never explicitly say what font display to be using 👍

**Comment by patrickhulce** [comment]
Filed #7342

**Comment by sdennett55** [comment]
I still get this even when `font-display: swap;` is the last declaration in an `@font-face` declaration.

**Comment by patrickhulce** [comment]
@sdennett55 what is your lighthouse version? This was fixed relatively recently.

**Comment by sdennett55** [comment]
@patrickhulce I'm on the most recent version of chrome, but to clarify, I don't have the issue here https://web.dev/ but I do in the audit tab of devtools.

Chrome: Version 73.0.3683.103 (Official Build) (64-bit)

**Comment by patrickhulce** [comment]
@sdennett55 stable Chrome runs very far behind current Lighthouse due to its release cadence. This has been fixed 👍

**Comment by chrisSowerby** [comment]
So how do you set the font-display if external CSS is bringing in fonts and you can't edit the CSS?

**Comment by rafiqullah0909** [comment]
> Not sure this is the reason for all the above failures, but I had a similar issue, so I checked and turns out that the code that Lighthouse uses to find a declared font-display property is `declaration.match(/font-display:(.*?);/);`. Please note the semicolon at the end of the regular expression. This means that if font-display is the last line in the CSS rule and does not end with a semicolon, the diagnostic audit fails. Given that this is a rather common practice, it might explain some of the above fails and should be addressed regardless.

Hello @Diklla 
Here is my Shopify Code for Font Setting. Could you please tell me. Where should I add "SEMICOLON" for solve the "Ensure text remains visible during Webfont load" Issue.


/* Typography */

{{ settings.heading_font | font_face: font_display: 'fallback' }}
{{ settings.text_font | font_face: font_display: 'fallback' }}

{%- assign text_font_bold = settings.text_font | font_modify: 'weight', 'bolder' -%}
{%- assign text_font_italic = settings.text_font | font_modify: 'style', 'italic' -%}
{%- assign text_font_bold_italic = text_font_bold | font_modify: 'style', 'italic' -%}

{{ text_font_bold | font_face: font_display: 'fallback' }}
{{ text_font_italic | font_face: font_display: 'fallback' }}
{{ text_font_bold_italic | font_face: font_display: 'fallback' }}

$heading-font-family : {{settings.heading_font.family}}, {{ settings.heading_font.fallback_families }};
$heading-font-weight : {{settings.heading_font.weight}};
$heading-font-style  : {{settings.heading_font.style}};
$heading-font-size   : {{settings.heading_size}};

$text-font-family : {{settings.text_font.family}}, {{ settings.text_font.fallback_families }};
$text-font-weight : {{settings.text_font.weight}};
$text-font-style  : {{settings.text_font.style}};

$uppercase-heading: {% if settings.uppercase_heading %}true{% else %}false{% endif %};

$base-text-font-size   : {{settings.base_text_font_size}}px;
$default-text-font-size: 14px;

Thanks
Rafiq

**Comment by antoinerousseau** [comment]
I'm still having this issue in web.dev too:
https://lighthouse-dot-webdotdevsite.appspot.com/lh/html?url=https://lebikini.netlify.com
Even though my font-faces all have `font-display:swap;` (you can see it directly at the top of the page source)
Any clue?

**Comment by patrickhulce** [comment]
@antoinerousseau those fonts it is flagging are indeed missing `font-display` on the network request, and I can't find the source you're referring to that has the `font-display` properly declared.
![image](https://user-images.githubusercontent.com/2301202/56473378-390ef480-6430-11e9-9287-89f75061cb94.png)

**Comment by antoinerousseau** [comment]
@patrickhulce I published an experiment since then, but you can see here what I was talking about:
https://5cbc80f7b4528a0008a0dbad--lebikini.netlify.com/
Look at the source or at the DevTools, it has `font-display:swap;` in all font-faces.

**Comment by patrickhulce** [comment]
Ah thanks for the repro URL @antoinerousseau!

**Comment by probablyup** [comment]
Same deal for https://lighthouse-dot-webdotdevsite.appspot.com/lh/html?url=https://sesamecare.com

If you look at the source for the site the `font-display` property is definitely set:

```
@font-face{font-family:'Gilroy';src:url('/assets/fonts/Gilroy-ExtraBold.woff2') format('woff2'),url('/assets/fonts/Gilroy-ExtraBold.woff') format('woff');font-display:fallback;font-weight:800;font-style:normal;} @font-face{font-family:'Gilroy';src:url('/assets/fonts/Gilroy-Bold.woff2') format('woff2'),url('/assets/fonts/Gilroy-Bold.woff') format('woff');font-display:fallback;font-weight:700;font-style:normal;} @font-face{font-family:'Gilroy';src:url('/assets/fonts/Gilroy-SemiBold.woff2') format('woff2'),url('/assets/fonts/Gilroy-SemiBold.woff') format('woff');font-display:fallback;font-weight:600;font-style:normal;} @font-face{font-family:'Gilroy';src:url('/assets/fonts/Gilroy-Medium.woff2') format('woff2'),url('/assets/fonts/Gilroy-Medium.woff') format('woff');font-display:fallback;font-weight:500;font-style:normal;} @font-face{font-family:'Gilroy';src:url('/assets/fonts/Gilroy-MediumItalic.woff2') format('woff2'),url('/assets/fonts/Gilroy-MediumItalic.woff') format('woff');font-display:fallback;font-weight:500;font-style:italic;} @font-face{font-family:'Gilroy';src:url('/assets/fonts/Gilroy-Regular.woff2') format('woff2'),url('/assets/fonts/Gilroy-Regular.woff') format('woff');font-display:fallback;font-weight:400;font-style:normal;}
```

**Comment by c0b41** [comment]
@probablyup follow up this issue https://github.com/GoogleChrome/lighthouse/issues/8493

**Comment by friska-nataniel** [comment]
Hi,

I encounter the same issue. Lighthouse audit shows 'Ensure text remains visible during webfont load' diagnostic. Right now, I have `font-display: fallback` on my code.
Here's the snippet of the [CSS file](https://www.static-src.com/frontend/static/css/init.93d22561ffcb7554b92d.css) per May 28th, 2020:
```
@font-face {
    font-display:fallback;font-family: effra;
    src: url(https://www.static-src.com/fonts/1.0.0/effra/effra-regular.woff2) format("woff2"),url(https://www.static-src.com/fonts/1.0.0/effra/effra-regular.ttf) format("truetype")
}

@font-face {
    font-display:fallback;font-family: efframedium;
    src: url(https://www.static-src.com/fonts/1.0.0/effra/effraMedium-regular.woff2) format("woff2"),url(https://www.static-src.com/fonts/1.0.0/effra/efframedium-regular.ttf) format("truetype")
}
```

Here's a few things that my team has tried:

1. Use local overrides to include `font-display: swap;` into vendor.css and init.css.
2. Check against regex used by Lighthouse to do this font display audit (refer ```
const
`fontDisplayMatch`
```in the [Lighthouse code](https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-core/audits/font-display.js)) and verified that `font-display: swap;` regardless of placement or whitespace should pass this audit.
3. Tested local overrides on LH versions 5.7.1 (in Chrome Stable) and version 6.0.0 (in Chrome Canary). Both have the same issue.
4. Testing on PageSpeed Insights is not applicable since it tests first view which is a different HTML and does not use custom fonts.

Hereby, I attached the lighthouse audit of  https://www.blibli.com. (refer to point number 4 where we can't test with PageSpeed Insights / [lighthouse-dot-webdotdevsite](https://lighthouse-dot-webdotdevsite.appspot.com/lh/html?url=https://www.blibli.com))

Your help would be really appreciated. Thank you.
[www.blibli.com-lighthouse-audit.zip](https://github.com/GoogleChrome/lighthouse/files/4693124/www.blibli.com-lighthouse-audit.zip)
